src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/DefaultLSPDocumentManagerTest.cs

```diff
@@ -69,10 +67,9 @@ public void TryGetDocument_UnknownDocument_ReturnsFalse()
         public void TryGetDocument_UntrackedDocument_ReturnsFalse()
         {
             // Arrange
-            var textView = Mock.Of<ITextView>();
             var manager = new DefaultLSPDocumentManager(JoinableTaskContext, UriProvider, LSPDocumentFactory);
-            manager.TrackDocumentView(TextBuffer, textView);
-            manager.UntrackDocumentView(TextBuffer, textView);
+            manager.TrackDocument(TextBuffer);
+            manager.UntrackDocument(TextBuffer);
```

Comment:
maybe I'm confused about what this is trying to accomplish, but what happens when we do
```c#
manager.TrackDocument(someTextBuffer);
manager.TrackDocument(someTextBuffer);
manager.TrackDocument(differentTextBuffer);

manager.UntrackDocument(someTextBuffer);
manager.UntrackDocument(someTextBuffer);
```

Is the idea that the RefCount is tracking the TOTAL number of file-tracking instances, rather than the count for a particular file?

Reply:
Nope, refcount is on a per-file basis. It means if you track the same text buffer multiple times we wont forget about it until you untrack it a similar number of times.